### ISSUE TITLE
errors in aishell2 finetune script

### ISSUE THREAD

**Issue by yondu22** [kaldi-asr/kaldi#4065]
missing fi after stage 2
also this assertion doesn't allow model training
````
LOG (nnet3-am-train-transitions[5.5.434~1-e167b]:main():nnet3-am-train-transitions.cc:131) Setting priors of pdfs in the model.
ASSERTION_FAILED (nnet3-am-train-transitions[5.5.434~1-e167b]:SetPriors():nnet3-am-train-transitions.cc:32) Assertion failed: (tmodel.NumPdfs() == am_nnet->NumPdfs())
```

**Comment by dophist** [comment]
in which script?

**Comment by yondu22** [comment]
https://github.com/kaldi-asr/kaldi/blob/master/egs/aishell2/s5/local/nnet3/tuning/finetune_tdnn_1a.sh

**Comment by dophist** [comment]
I see, will have a look. In the meantime, chain system is recommended over nnet3, make sure your development is on the right direction.

**Comment by yondu22** [comment]
yes, i just foudn this script on your folder and it looks compact and easy, i thought it was like this
original aishell model = exp/chain/nnet3
origina asihell data folder = data
new data folder = new_data
new model folder new_model

and then run it and it uses the old model trained with run.sh to build fine tune with the new data and output a new model

**Comment by underdogliu** [comment]
@yondu22 Like @dophist said, it might be good to use chain model directly instead of nnet3 - I do not think fine-tuning nnet3 will make nnet3 model advantageous over chain-trained model. Chain training script should be efficient as well, if not better (should be better). nnet3 model, however, can be used to generate slightly better alignments for chain training, although it doubles the training time.

**Comment by naxingyu** [comment]
@dophist @underdogliu  We should fix the script bug anyway. And regarding the assertion, make sure you used the same tree in align model and seed nnet3 model.

**Comment by yondu22** [comment]
yes chain is fine to use, i just saw this script was trying to use it, if you make a chain one that is great, ill use that

**Comment by kkm000** [comment]
@naxingyu 
> @dophist @underdogliu We should fix the script bug anyway.

Is anyone looking into it?

**Comment by dophist** [comment]
@kkm000 I'll submit a trivial PR to fix it this weekend.

**Comment by kkm000** [comment]
@dophist, amazing, thank you!

**Comment by Asma-droid** [comment]
Any solution please for this issue?

**Comment by kkm000** [comment]
@Asma-droid, it has been fixed in #4179. Is your Kaldi too old? If not, please post your repro in a new issue, don't assume it's the same issue, unless you have confirmed that to near certainty. Thanks!
